src/main/java/uk/gov/justice/hmpps/prison/service/LocationService.java

```diff
@@ -100,6 +100,24 @@ public Location getLocation(final long locationId, boolean includeInactive) {
             .orElseThrow(EntityNotFoundException.withId(locationId));
     }
 
+    public Optional<Location> getLocationByDescription(final String description) {
+        return agencyInternalLocationRepository.findOneByDescription(description)
+            .map(
+                l -> LocationProcessor.processLocation(Location.builder()
```

Comment:
Not too sure about the parameter name !

Reply:
I know, but that is what it is called in the DB and the repo

Reply:
@steverendell what do you think ?  code, id, locationCode, Basically it is the description field that happens to be unique

Reply:
Anything like code or location is fine, its just that an 'l' looks like a '1' .. and is hard to spot in the block too 🙂